cli/azd/pkg/infra/provisioning/manager.go

```diff
@@ -31,7 +31,21 @@ type Manager struct {
 	options             *Options
 }
 
+// defaultOptions for this package.
+const (
+	defaultModule = "main"
+	defaultPath   = "infra"
+)
+
 func (m *Manager) Initialize(ctx context.Context, projectPath string, options Options) error {
+	// applied defaults if missing
+	if options.Module == "" {
+		options.Module = defaultModule
+	}
+	if options.Path == "" {
+		options.Path = defaultPath
+	}
+
```

Comment:
Why do we need defaults at the manager level when the underlying providers also have their own defaults?  This means that the provider level defaults would never get set because the manager is setting them to its own value before `Initialize` is ever called on the provider.

Reply:
that's for keeping packages independent to each other (no coupling).

If I want to use the project package, I shouldn't need to depend on the provisioning package indirectly. 

Provider-level defaults are used when the provider is consumed by a component/package where those values are not defined.

Having non-coupling components makes it all easy. Replacing/updating underlaying components, for example.